File: SWEA/D2/1959. 두 개의 숫자열/두 개의 숫자열.py
T = int(input())
for t in range(T):
    n, m = map(int, input().split())
    a = list(map(int, input().split()))
    b = list(map(int, input().split()))
     
    result = -2100
     
    # 긴 배열을 기준으로 짧은 배열을 움직이므로, 어느 배열이 긴지 기억해야 한다.
    # n, a 가 무조건 긴거라고 한다면
    #print('입력 정상')
    if m > n: 
        n, m = m, n
        a, b = b, a
    #print('바꾸기 정상')
    # 이제 n, a가 무조건 긴 배열이다. a를 기준으로 b를 움직이면 된다.
    for i in range(0, n-m+1):    # 시작위치 조정 0~(n-m)
        # 곱하는 것은 시작 index부터 짧은 배열 b의 길이 m만큼
        tmp = 0
        #print('index 변경 정상')
        for s in range(i, i+m, 1):
            # 짧은 배열의 index는 증가하면 안 된다. 계속 0~m 이어야 한다.
            tmp += a[s]*b[s-i]
            #print(tmp, s, a[s], b[s-i])
            #print('곱셈 후 더하기 정상')
        if tmp > result: result = tmp
     
    print(f'#{t+1} {result}')

File: SWEA/D2/1959. 두 개의 숫자열/두 개의 숫자열.java
import java.util.*;
import java.io.*;
 
class Solution
{
    public static void main(String args[]) throws Exception
    {
       
        Scanner sc = new Scanner(System.in);
        int T;
        T=sc.nextInt();
        
        for(int test_case = 1; test_case <= T; test_case++)
        {
         
            int n, m;
            n = sc.nextInt();
            m = sc.nextInt();
            int[] a = new int[n];
            int[] b = new int[m];
            for (int i = 0; i<n; i++) {
                a[i] = sc.nextInt();
            }
            for (int i=0; i<m; i++) {
                b[i] = sc.nextInt();
            }
             
            int big;
            big = (n >= m) ? n : m;
            int[] A = new int[big];
            int[] B = new int[big];
            int x, y;
             
            if (m > n) {
                x = m;
                y = n;
                A = b.clone();
                B = a.clone();
            } else {
                x = n;
                y = m;
                A = a.clone();
                B = b.clone();
            }
             
            // x: 둘 중 긴 배열의 길이, A: 둘 중 긴 배열의 배열, y: 짧은 배열의 길이, B: 짧은 배열
            // 이제 곱하고 더하면 된다.
            int result = -2100000000;
            for (int i=0; i<=x-y; i++) {
                int tmp = 0;
                for (int s=i; s<i+y; s++) {
                    tmp += A[s]*B[s-i];
                }
                if (tmp > result) result = tmp;
            }
             
            System.out.printf("#%d %d\n", test_case, result);
 
        }
    }
}